terraform/devnet-0/digitalocean.tf

```diff
@@ -0,0 +1,296 @@
+////////////////////////////////////////////////////////////////////////////////////////
+//                                        VARIABLES
+////////////////////////////////////////////////////////////////////////////////////////
+variable "digitalocean_project_name" {
+  type    = string
+  default = "4844 Testnets"
+}
+
+variable "digitalocean_ssh_key_name" {
+  type    = string
+  default = "barnabasbusa"
+}
+
+variable "digitalocean_regions" {
+  default = [
+    "nyc1",
+    "sgp1",
+    "lon1",
+    "nyc3",
+    "ams3",
+    "fra1",
+    "tor1",
+    "blr1",
+    "sfo3",
+    "syd1"
+  ]
+}
+
+////////////////////////////////////////////////////////////////////////////////////////
+//                                        LOCALS                                      
+////////////////////////////////////////////////////////////////////////////////////////
+locals {
+  digitalocean_base_cidr_block = var.digitalocean_base_cidr_block
+  digitalocean_vpcs = {
+    for region in var.digitalocean_regions : region => {
+      name     = "${var.ethereum_network}-${region}"
+      region   = region
+      ip_range = cidrsubnet(local.base_cidr_block, 8, index(var.digitalocean_regions, region))
+    }
+  }
+}
+
+locals {
+  digitalocean_vm_groups = flatten([
+    for vm_group in local.vm_groups :
+    [
+      for i in range(0, vm_group.count) : {
+        group_name = "${vm_group.name}"
+        id         = "${vm_group.name}-${i + 1}"
+        vms = {
+          "${i + 1}" = {
+            tags   = "group_name:${vm_group.name},val_start:${vm_group.validator_start + (i * (vm_group.validator_end - vm_group.validator_start) / vm_group.count)},val_end:${min(vm_group.validator_start + ((i + 1) * (vm_group.validator_end - vm_group.validator_start) / vm_group.count), vm_group.validator_end)}"
+            region = element(var.digitalocean_regions, i % length(var.digitalocean_regions))
+            size   = try(vm_group.size, local.digitalocean_default_size)
+          }
+        }
+      }
+    ]
+  ])
+}
+
+locals {
+  digitalocean_default_region = "ams3"
+  digitalocean_default_size   = "c-4"
```

Comment:
drop the default size to some tiny node for the template? so even by mistake it won't cost much

Reply:
c-4 is already very small, changed to c-2